### PR TITLE
fix(fmt, lint): not print the number of checked files when quiet option is specified

### LINKED ISSUE #7463: Deno 1.4.0 fmt --quiet option regression
The quiet option is not honoured:

```
$ deno --version
deno 1.3.3
v8 8.6.334
typescript 4.0.2

$ deno fmt --quiet Drakefile.ts 

$
```

```
$ deno --version
deno 1.4.0
v8 8.7.75
typescript 4.0.2

$ deno fmt --quiet Drakefile.ts 
Checked 1 file

$
```

The `Checked N file` message appears to be new (`fmt` was quiet by default in Deno 1.3.3).

### PR DESCRIPTION
**Pull request by magurotuna**
This PR fixes `quiet` flag regression reported in #7463.

The main idea of this fix is that using logger macros instead of `eprintln!` and `println!`. 
The global logger knows whether `quiet` option is appended, so we don't need to take care of the flag in each subcommand implementation.

While doing my work I ran into a problem; when I used `log::error!` macro, the output was prefixed with `ERROR RS -`. I thought it was weird, because for JS or TS, I didn't get such output. Actually I got `DEBUG JS - ` prefixed outputs in debug level log though.

The current setting of `env_logger` adds `<LogLevel> RS -` prefix if a log level is smaller than `LogLevel::Info`. In Rust, the log levels that are smaller than `Info` are `Error` and `Warn`. This is why I got `ERROR RS -` prefixed outputs.
I'm not really sure, but it would be better to add prefixes for `Debug` and `Trace` level logs (bacause JS and TS do so), so I changed the direction of the inequality sign.

Fixes #7463

### CODE REVIEW

File: cli/lint.rs [hunk 1]
```diff
@@ -105,9 +105,9 @@ pub async fn lint_files(
 pub fn print_rules_list() {
   let lint_rules = rules::get_recommended_rules();
 
-  println!("Available rules:");
+  info!("Available rules:");
   for rule in lint_rules {
-    println!(" - {}", rule.code());
+    info!(" - {}", rule.code());
```
Comment (bartlomieju) [inline cli/lint.rs:110]: This change makes no sense - `deno lint --unstable --rules` is only used to print out a list of rules, if we apply `info!()` here then you might not get output at all

File: cli/tests/performance_stats.out [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+DEBUG RS - [WILDCARD] - Identifiers: [WILDCARD]
+DEBUG RS - [WILDCARD] - Symbols: [WILDCARD]
+DEBUG RS - [WILDCARD] - Types: [WILDCARD]
+DEBUG RS - [WILDCARD] - Instantiations: [WILDCARD]
+DEBUG RS - [WILDCARD] - Parse time: [WILDCARD]
+DEBUG RS - [WILDCARD] - Bind time: [WILDCARD]
+DEBUG RS - [WILDCARD] - Check time: [WILDCARD]
+DEBUG RS - [WILDCARD] - Emit time: [WILDCARD]
+DEBUG RS - [WILDCARD] - Total TS time: [WILDCARD]
+DEBUG RS - [WILDCARD] - Compile time: [WILDCARD]
```
Comment (bartlomieju) [inline cli/tests/performance_stats.out:13]: What's the reason for this change?

File: cli/tests/type_directives_js_main.js.out [hunk 3]
```diff
@@ -1,3 +1,3 @@
 [WILDCARD]
-fetch_source_file specifier: file:[WILDCARD]cli/tests/subdir/type_reference.d.ts [WILDCARD]
-[WILDCARD]
\ No newline at end of file
+DEBUG RS - [WILDCARD] - fetch_source_file specifier: file:[WILDCARD]cli/tests/subdir/type_reference.d.ts [WILDCARD]
```
Comment (bartlomieju) [inline cli/tests/type_directives_js_main.js.out:2]: Ditto

File: cli/lint.rs (hunk 1 above)
Comment (magurotuna) [reply cli/lint.rs:110]: No, it does work. `deno lint --unstable --rules` prints the rules, except that it goes to stderr (not stdout).
If we add `--rules` and `--quiet` options at the same time, then we will get nothing.

image

Comment (bartlomieju) [reply cli/lint.rs:110]: @magurotuna that's exactly my problem `--rules --quiet` makes no sense - in that case rules should still be printed - that's the only use case for this flag.

File: cli/tests/performance_stats.out (hunk 2 above)
Comment (magurotuna) [reply cli/tests/performance_stats.out:13]: If I didn't change it, the tests would fail. Because my patch adds `DEBUG RS -` prefixes to logs with debug level. Previously such prefixes were not there, but they are added in my implementation.

File: cli/lint.rs (hunk 1 above)
Comment (magurotuna) [reply cli/lint.rs:110]: Oh, I got it. You're right. I'll fix it to use `println!` here.

Comment (magurotuna) [reply cli/lint.rs]: Done!

**Review by bartlomieju — CHANGES_REQUESTED** [review]
@magurotuna thank you for the PR, but I'm afraid something is off - changing debug output seems like a big change. Could you split it into two separate PRs - one for fixing `fmt`/`lint` output and second for fixing logging output?

**Comment by magurotuna** [comment]
@bartlomieju Sure, I'll do that. I won't use logger macros on the first PR for `fmt` and `lint` fix, just deal with `quiet` flag in each subcommand implementation.

**Comment by bartlomieju** [comment]
> @bartlomieju Sure, I'll do that. I won't use logger macros on the first PR for `fmt` and `lint` fix, just deal with `quiet` flag in each subcommand implementation.

@magurotuna I think you can safely use `info!` macro and just use `eprintln!` for actual errors

**Comment by magurotuna** [comment]
@bartlomieju ah, that's right. I'll do that way.

**Comment by magurotuna** [comment]
Close this because I opened two separated PRs: #7579, #7582
